### ISSUE TITLE
create command line input prompt

### ISSUE THREAD

**Issue by mimmerma** [mimmerma/familyTree#1]
- using built-in raw_input() method
- explored cmd.Cmd module for more advanced user interface options, deemed unnecessary
- import readline module to enable bash-like functionality for user

**Comment by jdimmerman** [comment]
There are many ways to format tickets, but a good idea to keep in mind is that you always want to end with 'Acceptance Criteria' that unambiguously define what must exist in order for the ticket to be completed (meet the 'Definition of Done').

See my next comment for my suggestion

**Comment by jdimmerman** [comment]
**Agile Statement**
As a user of the Family Tree commandline application, I want to be able to enter the appropriate inputs to add a Person so that I have a method by which I can define a new Person when that part of the program is implemented.

**Plain English Statement**
As a first step to building the Family Tree program, we should create a way to enter the arguments that will ultimately allow the user to add a person.  This will include the proper mechanism by which we can parse the input arguments.  Note that this will _not_ include either validation or actually implementation of adding a person, just the code to parse the inputs themselves.

**Implementation Notes**
- This could be achieved using the built in python raw_input() method
- This could be achieved using the cmd.Cmd more advanced  user interface options.  After further research, this has been deemed unnecessary
- You will likely need to import the readline module to enable bash-like interaction from the commandline user

**Acceptance Criteria (AC)**
* The user is able to call the familyTree commandline application and pass in several arguments
* The arguments are parsed and printed to the commandline in order, along with the names of the variables they are stored in, to confirm that the parsing method works.
* The program then calls a method to validate input, which throws a NotImplemented exception (implementation of this will be part of a later tickete)
* Unit tests exist for the parsing function if possible

**Comment by mimmerma** [comment]
1. In Agile Statement, I was originally confused with the use of "method",
but I think in this context it is synonymous with "means"

2.  In Plain English Statement, when you say "parse", do you mean that this
step will assign the inputs to a variable and then also divide the string
into it's constituent parts?  Because I thought I would keep it in very
small bits, and just build a basic prompt in this step and assign the user
input (in response to the prompt) to a variable.  But I could also just
parse it in this step pretty easily, but see #3.

3. Is it preferable if I have the user input a string with multiple
attributes, which is then parsed (maybe using RegEx), or to have the
interface loop through a sequence of prompts which different questions to
obtain a single attribute at a time (requiring no parsing).  The former
approach would have optional instance variables with default values for the
Person class, whereas the latter would require the user to enter something
(if only the empty string) for each attribute.

4.  In Acceptance Criteria, is it necessary to include the reference to a
validation method, or could I just put that straight into another ticket?

5. Do you always have "this will include and not include" in the Plain
English Statement but not the Agile Statement, and the Agile Statement is
from the user's point of view (first person singular) but the Plain English
Statement is from the programmer's point of view (first person plural)?

6.  Is it important to include references in this ticket for future program
features or functionality, such as when you say "will ultimately allow the
user to add a person" under Plain English Statement?

**Comment by jdimmerman** [comment]
1) Cool yeah makes sense

2) I was just giving an example - feel free to alter the scope of this ticket (scope == what the results of the ticket are meant to accomplish) yourself.

3) I always recommend getting away from user prompts.  Prompting the user means that you can't call the script from automation and also makes the end product very un-parseable.  So maybe a generic call would be:
`familyTree.py [method] [<optional-argument>] `
So an example
`familyTree.py addPerson "Jason Immerman" 05221988`
`familyTree.py makeChild "Jason Immerman" "Irene Immerman,Michael Immerman"`

4) Either, as long as its clear and easy for everyone to know what you mean and find what's important

5) The plain english part is purely an example - again, just the info for everyone to be on the same page about pertinent info and such. There aren't really any standards.  The Agile Statement is definitely a strict standard but most places don't follow it; and yes, it's from the user's point of view

6) Generally, either. Personally, I prefer references to the tickets but I'm an overplanner in all parts of life
